src/audio/dai.c

```diff
@@ -215,6 +216,12 @@ static int dai_comp_get_hw_params(struct comp_dev *dev,
 		return ret;
 	}
 
+	/* overwrite frame_fmt hardware parameters as DAI component is able
+	 * to convert stream with different frame_fmt's
+	 */
+	if (dd->frame_fmt)
+		params->frame_fmt = dd->frame_fmt;
+
```

Comment:
@bkokoszx can you clarify for me the commit message and comment above "DAI component is able to convert stream with different frame_fmt's"?
It was my understanding that the DAI operates with a fixed format, and conversions are e.g. handled in a volume component. What am I missing?

Signed-off-by: Bartosz Kokoszko &lt;bartoszx.kokoszko@linux.intel.com&gt;</pre>

Reply:
@plbossart 
For some time the DAI component has been able to make ```frame_fmt``` conversion. We do not need volume component to do that. This capability has been added in PR #2029.

Reply:
@bkokoszx best to add more context to this comment message. This is about back propagating the DAI format back through the pipeline so that any converter component can convert from format X -> Y if needed.
I would also make this a static inline since it will be used as part of runtime by multi-DAI support.